editor/editor_data.cpp

```diff
@@ -947,9 +947,9 @@ Variant EditorData::script_class_instance(const String &p_class) {
 	if (ScriptServer::is_global_class(p_class)) {
 		Ref<Script> script = script_class_load_script(p_class);
 		if (script.is_valid()) {
-			Object *obj = ClassDB::instantiate(script->get_instance_base_type());
+			Variant obj = ClassDB::instantiate(script->get_instance_base_type());
 			if (obj) {
-				obj->set_script(script);
+				obj.operator Object *()->set_script(script);
```

Comment:
That looks pretty confusing, and warrants at least an explanatory comment.
Are you also sure that `if (obj)` will still work as intended now that it's checking a Variant instead of a pointer?

Reply:
you're right, it is confusing, i'll add a comment
this was just reverted to what it was before (the Variant and cast part seemed weird and confusing and unecessary, so i changed it):
https://github.com/godotengine/godot/blob/c7ceb94e372216b1b033d7c2ac26d5b7545c4dac/editor/editor_data.cpp#L948-L952
and yes it will just do a null check:
https://github.com/godotengine/godot/blob/c7ceb94e372216b1b033d7c2ac26d5b7545c4dac/core/variant/variant.cpp#L933-L935